packages/@cdktf/cli-core/src/lib/cdktf-project.ts

```diff
@@ -618,11 +624,10 @@ export class CdktfProject {
 
     // We end the loop when all stacks are started, now we need to wait for them to be done
     // We wait for all work to finish even if one of the promises threw an error.
-    const currentWork = this.stacksToRun
-      .filter((ex) => ex.currentWorkPromise)
-      .map((ex) => ex.currentWorkPromise);
-
-    await ensureAllSettledBeforeThrowing(Promise.all(currentWork), currentWork);
+    await ensureAllSettledBeforeThrowing(
```

Comment:
An error on leads to the workingPromise to be cleared, so this is a better way to ensure no work is left in progress.